libs/moment-datetime-adapter/adapter/moment-datetime-adapter.ts

```diff
@@ -0,0 +1,312 @@
+
+        return range(24, (i) => {
+            return this.clone(momentDate).hour(i).format(format);
+        });
+    };
+
+    getMinuteNames({ twoDigit }: { twoDigit: boolean }): string[] {
+        const format: string = twoDigit ? 'mm' : 'm';
+        const momentDate = this._createMomentDate();
+
+        return range(60, (i) => {
+            return this.clone(momentDate).minute(i).format(format);
+        });
+    };
+
+    getSecondNames({ twoDigit }: { twoDigit: boolean }): string[] {
+        const format: string = twoDigit ? 'ss' : 's';
+        const momentDate = this._createMomentDate();
+
+        return range(60, (i) => {
+            return this.clone(momentDate).second(i).format(format);
+        });
+    };
+
+    getDayPeriodNames(): [string, string] {
+        console.log('here');
+
+        return ['', ''];
+    };
+
+    getFirstDayOfWeek(): number {
+        return this._localeData.firstDayOfWeek;
+    };
+
+    getNumDaysInMonth(date: Moment): number {
+        return this.clone(date).daysInMonth();
+    };
+
+    parse(value: any, parseFormat: MomentFormatSpecification = ''): Moment | null {
+        if (value && typeof value === 'string') {
+            return this._createMomentDate(value, parseFormat, this.locale);
+        }
+
+        return value ? this._createMomentDate(value).locale(this.locale) : null;
+    };
+
+    format(date: Moment, displayFormat: string): string {
+        date = this.clone(date);
+
+        if (!this.isValid(date)) {
+            throw Error('MomentDateTimeAdapter: Cannot format invalid date.');
+        }
+
+        return date.format(displayFormat);
+    };
+
+    createDate(year: number, month: number, date: number): Moment {
+        if (month < 0 || month > 11) {
+            throw Error(`Invalid month index "${ month }". Month index has to be between 0 and 11.`);
+        }
+
+        if (date < 1) {
+            throw Error(`Invalid date "${ date }". Date has to be greater than 0.`);
+        }
+
+        const result = this._createMomentDate({ year: year, month: month, date: date }).locale(this.locale);
+
+        if (!result.isValid()) {
+            throw Error(`Invalid date "${ date }" for month with index "${ month }".`);
+        }
+
+        return result;
+    };
+
+    today(): Moment {
+        return this._createMomentDate()
+            .locale(this.locale)
+            .startOf('day');
+    };
+
+    now(): Moment {
+        return this._createMomentDate().locale(this.locale);
+    };
+
+    addCalendarYears(date: Moment, years: number): Moment {
+        return this.clone(date).add({ years: years });
+    };
+
+    addCalendarMonths(date: Moment, months: number): Moment {
+        return this.clone(date).add({ months: months });
+    };
+
+    addCalendarDays(date: Moment, days: number): Moment {
+        return this.clone(date).add({ days: days });
+    };
+
+    getAmountOfWeeks(year: number, month: number, firstDayOfWeek: number): number {
+        return null;
```

Comment:
We can calculate it the same way as in the fd-datetime-adapter.
Actually we could move it in the base class.

Reply:
Replaced to the base class